cmd/gen-go-sample/main.go

```diff
@@ -224,6 +225,77 @@ func (g *generator) genSample(ifaceName string, methConf GAPICMethod, regTag str
 	}
 	g.imports[inSpec] = true
 
+	initInfo, err := g.getInitInfo(inType, methConf, valSet)
+	argNames := initInfo.argNames
+	argTrees := initInfo.argTrees
+	flagNames := initInfo.flagNames
+	files := initInfo.files
+	itree := initInfo.reqTree
+
+	if err := g.sampFuncAndClient(serv.GetName(), meth.GetName(), initInfo, regTag); err != nil {
```

Comment:
A suggestion: How about refactoring this function slightly?:
```
args, err := getFunctionArgList(...)
p.(...) // function signature, region tag, and client
 ...
err := setUpRequest(...)
err := processResponse(...)
p.(...) // end function, end region tag, etc.
  ...
return writeMain(...)
```
(not showing the error handling above, obvs)

This makes it easier to grok the high-level structure, while delegating groups of details to functions, and we get the symmetry of opening and closing the sample function and region tags in the same function. WDYT?

Reply:
SG.